src/eh1/pin.rs

```diff
@@ -199,16 +221,73 @@ impl InputPin for Mock {
     }
 }
 
+/// Single digital output pin that remembers its state and can be toggled between high and low states
+impl StatefulOutputPin for Mock {
+    /// Toggle the pin low to high or high to low
+    fn toggle(&mut self) -> Result<(), Self::Error> {
+        let Transaction { kind, err } = self.next().expect("no expectation for pin::toggle call");
+
+        assert_eq!(kind, TransactionKind::Toggle, "expected pin::toggle");
+
+        match err {
+            Some(e) => Err(e),
+            None => Ok(()),
+        }
+    }
+
+    /// Is the output pin set high?
+    fn is_set_high(&mut self) -> Result<bool, Self::Error> {
+        let mut s = self.clone();
+
+        let Transaction { kind, err } = s.next().expect("no expectation for pin::is_set_high call");
+
+        assert_eq!(
+            kind,
+            TransactionKind::GetSetState(State::High),
+            "expected pin::is_set_high"
+        );
+
+        if let Some(e) = err {
+            Err(e)
+        } else if let TransactionKind::GetSetState(v) = kind {
+            Ok(v == State::High)
+        } else {
+            unreachable!();
+        }
```

Comment:
Hmm, I'm not sure if this makes sense. In the `assert_eq!` above, you're asserting that the transaction is `GetSetState(High)`. This already implies that the state is high, so the check at 252 doesn't do anything meaningful.

On the other hand, shouldn't this transaction work both for high and low states? In the case of a low state, it would simply return `Ok(false)`.

I suspect that the `assert_eq!` needs to be replaced with something like `assert!(matches!(kind, TransactionKind::GetSetState(_)))` and an appropriate error message. Same thing for `is_set_low`. In that case, no matter whether the driver calls `is_set_high` or `is_set_low`, it will work (and return either true or false).

Reply:
Oh yeah, good catch! Thanks. I've added a `fixup` commit to fix it and added some unit tests for it too.